irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -159,6 +163,43 @@ namespace iroha {
     });
   }
 
+  bool PendingTransactionStorageImpl::isReplay(shared_model::interface::TransactionBatch const &batch) {
+    auto cache_ptr = presence_cache_.lock();
+    assert(!!cache_ptr);
```

Comment:
if the pointer is weak, then you suppose its object can be destroyed before it, so this assertion may detonate, but only in testing build. let's place something like
```c++
if (not cache_ptr) return false;
```
after the assertion to avoid dirty crashes in release build.

Reply:
As it was discussed in private, assert should be enough in current place.